packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/FLTCam.m

```diff
@@ -1302,9 +1303,50 @@ - (BOOL)setupWriterForPath:(NSString *)path {
   return YES;
 }
 
+// configure application wide audio session manually to prevent overwriting
+// flag MixWithOthers by capture session, only change category if it is considered
+// as upgrade which means it can only enable ability to play in silent mode or
+// ability to record audio but never disables it, that could affect other plugins
+// which depend on this global state, only change category or options if there is
+// change to prevent unnecessary route changes which can cause lags
```

Comment:
> unnecessary route changes

What is a "route change"?

Reply:
https://developer.apple.com/documentation/avfaudio/responding_to_audio_route_changes?language=objc

When category changes it can cause route change seems:

https://developer.apple.com/documentation/avfaudio/avaudiosessionroutechangereason/avaudiosessionroutechangereasoncategorychange?language=objc

Anyway the effect of setting `AVAudioSessionCategoryPlayAndRecord` multiple times even when the category is unchanged and there are even no headphones causes half of second or longer silence on my device. Also it causes the video_player plugin to pause and must be manually called `play` again.